idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/quotes/cache/QuoteCacheService.kt

```diff
@@ -0,0 +1,48 @@
+package arrow.meta.ide.plugins.quotes.cache
+
+import arrow.meta.ide.dsl.application.services.Id
+import arrow.meta.ide.dsl.application.services.IdService
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtFile
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * a QuoteCache implementation as a IdService, which stores Quote transformations on disk.
+ * This is an initial implementation, as it does not populate the index of other connected projects,
+ * and the cache stored on disk get out of sync with external changes to the disk, projects sharing source files or other situations.
+ * fixme cache both per module? modules may define different ktFiles for the same package fqName
+ * TODO: register this via Meta.
+ */
+private object QuoteCacheService : IdService<ConcurrentHashMap<KtFile, QuoteInfo>>, QuoteCache {
```

Comment:
It's an object, but it's registered as a project service. 
There's one service instance per project, so it should be a `class`, not an `object`.